test_can_generate_interval_endpoints is flaky

```
=================================== FAILURES ===================================
_ test_can_generate_interval_endpoints[-2.22507385851e-308-2.22507385851e-308] _
[gw1] linux2 -- Python 2.7.14 /usr/bin/python2
Traceback (most recent call last):
  File "/build/python-hypothesis/src/hypothesis-python-3.44.16-py2/tests/cover/test_float_nastiness.py", line 84, in test_can_generate_interval_endpoints
    find(interval, lambda x: x == r, settings=settings(max_examples=10000))
  File "/build/python-hypothesis/src/hypothesis-python-3.44.16-py2/src/hypothesis/core.py", line 1128, in find
    raise NoSuchExample(get_pretty_function_description(condition))
NoSuchExample: No examples found of condition lambda x: x == r
=============================== warnings summary ===============================
tests/numpy/test_gen_data.py::test_can_minimize_float_arrays
  /usr/lib/python2.7/site-packages/numpy/core/_methods.py:32: RuntimeWarning: invalid value encountered in reduce
    return umr_sum(a, axis, dtype, out, keepdims)

-- Docs: http://doc.pytest.org/en/latest/warnings.html
======== 1 failed, 2380 passed, 6 skipped, 1 warnings in 100.77 seconds ========
```

Environment:
```
============================= test session starts ==============================
platform linux2 -- Python 2.7.14, pytest-3.3.2, py-1.5.0, pluggy-0.6.0
rootdir: /build/python-hypothesis/src/hypothesis-python-3.44.16-py2, inifile: tox.ini
plugins: hypothesis-3.44.16, xdist-1.22.0, forked-0.2, flaky-3.4.0
```

--- a/tests/cover/test_float_nastiness.py
+++ b/tests/cover/test_float_nastiness.py
@@ -78,6 +78,7 @@ def test_does_not_generate_positive_if_right_boundary_is_negative(x):
     (-1.0, 0.0),
     (-sys.float_info.min, sys.float_info.min),
 ])
+@flaky(max_runs=4, min_passes=1)
 def test_can_generate_interval_endpoints(l, r):
     interval = st.floats(l, r)
     find(interval, lambda x: x == l, settings=settings(max_examples=10000))

--- a/tests/quality/test_shrink_quality.py
+++ b/tests/quality/test_shrink_quality.py
@@ -88,13 +88,11 @@ def test_minimize_3_set_of_tuples():
         lambda x: len(x) >= 2) == set(((0,), (1,)))
 
 
-@pytest.mark.parametrize('seed', [11, 43, 49, 56, 65])
-def test_minimize_sets_of_sets(seed):
+def test_minimize_sets_of_sets():
     elements = integers(1, 100)
     size = 8
     set_of_sets = minimal(
         sets(frozensets(elements), min_size=size),
-        random=Random(seed),
     )
     assert frozenset() in set_of_sets
     assert len(set_of_sets) == size

--- a/tests/quality/test_shrink_quality.py
+++ b/tests/quality/test_shrink_quality.py
@@ -91,9 +91,7 @@ def test_minimize_3_set_of_tuples():
 def test_minimize_sets_of_sets():
     elements = integers(1, 100)
     size = 8
-    set_of_sets = minimal(
-        sets(frozensets(elements), min_size=size),
-    )
+    set_of_sets = minimal(sets(frozensets(elements), min_size=size))
     assert frozenset() in set_of_sets
     assert len(set_of_sets) == size
     for s in set_of_sets:

--- a/tests/quality/test_shrink_quality.py
+++ b/tests/quality/test_shrink_quality.py
@@ -88,11 +88,13 @@ def test_minimize_3_set_of_tuples():
         lambda x: len(x) >= 2) == set(((0,), (1,)))
 
 
-def test_minimize_sets_of_sets():
+@pytest.mark.parametrize('seed', [11, 43, 49, 56, 65])
+def test_minimize_sets_of_sets(seed):
     elements = integers(1, 100)
-    size = 10
+    size = 8
     set_of_sets = minimal(
-        sets(frozensets(elements)), lambda s: len(s) >= size
+        sets(frozensets(elements), min_size=size),
+        random=Random(seed),
     )
     assert frozenset() in set_of_sets
     assert len(set_of_sets) == size

